wp-admin/js/widgets/media-video-widget.js

```diff
@@ -0,0 +1,236 @@
+			}
+
+			// If the external video is hosted elsewhere, return default icon.
+			// TODO: Get markup/image for generic thumbnail
+			return $.Deferred().resolveWith( control, [ { thumbnail_url: '' } ] ).promise();
+		},
+
+		/**
+		 * Get YouTube video ID from URL.
+		 *
+		 * @param {string} url URL from model.
+		 * @returns {string} YouTube video ID
+		 */
+		_getYouTubeIdFromUrl: function _getYouTubeIdFromUrl( url ) {
+			var urlParts;
+			urlParts = url.split( /(vi\/|v=|\/v\/|youtu\.be\/|\/embed\/)/ );
+			return ! _.isUndefined( urlParts[2] ) ? urlParts[2].split( /[^0-9a-z_\-]/i )[0] : urlParts[0];
+		},
+
+		/**
+		 * Get the instance props from the media selection frame.
+		 *
+		 * @param {wp.media.view.MediaFrame.Select} mediaFrame Select frame.
+		 * @returns {object} Props
+		 */
+		getSelectFrameProps: function getSelectFrameProps( mediaFrame ) {
+			var control = this,
+				state = mediaFrame.state(),
+				props = {};
+
+			if ( 'embed' === state.get( 'id' ) ) {
+				props = control._getEmbedProps( mediaFrame, state.props.toJSON() );
+			} else {
+				props = control._getAttachmentProps( mediaFrame, state.get( 'selection' ).first().toJSON() );
+			}
+
+			return props;
+		},
+
+		/**
+		 * Get the instance props from the media selection frame.
+		 *
+		 * @param {wp.media.view.MediaFrame.Select} mediaFrame Select frame.
+		 * @param {object} attachment Attachment object.
+		 * @returns {object} Props
+		 */
+		_getAttachmentProps: function _getAttachmentProps( mediaFrame, attachment ) {
+			var props = {}, displaySettings;
+
+			displaySettings = mediaFrame.content.get( '.attachments-browser' ).sidebar.get( 'display' ).model.toJSON();
+			if ( ! _.isEmpty( attachment ) ) {
+				_.extend( props, {
+					attachment_id: attachment.id,
+					caption: attachment.caption,
+					description: attachment.description,
+					link_type: displaySettings.link,
+					url: attachment.url
+				} );
+			}
+
+			return props;
+		},
+
+		/**
+		 * Get the instance props from the media selection frame.
+		 *
+		 * @param {wp.media.view.MediaFrame.Select} mediaFrame Select frame.
+		 * @param {object} attachment Attachment object.
+		 * @returns {object} Props
+		 */
+		_getEmbedProps: function _getEmbedProps( mediaFrame, attachment ) {
+			var props = {};
+
+			if ( ! _.isEmpty( attachment ) ) {
+				_.extend( props, {
+					attachment_id: 0,
+					caption: attachment.caption,
+					description: attachment.description,
+					link_type: attachment.link,
+					url: attachment.url
+				} );
+			}
+
+			return props;
+		},
+
+		/**
+		 * Open the media image-edit frame to modify the selected item.
+		 *
+		 * @returns {void}
+		 */
+		editMedia: function editMedia() {
+			var control = this, mediaFrame, metadata, updateCallback;
+
+			metadata = {
+				attachment_id: control.model.get( 'attachment_id' ),
+				caption: control.model.get( 'caption' ),
+				description: control.model.get( 'description' ),
+				link_type: control.model.get( 'link_type' )
```

Comment:
There are missing properties being passed from the model into the media frame. For example, `autoplay` and `loop`.